Source/WebCore/page/ImageAnalysisQueue.cpp

```diff
@@ -158,10 +159,19 @@ void ImageAnalysisQueue::resumeProcessing()
                 m_pendingRequestCount--;
 
             resumeProcessingSoon();
+
+            if (m_queue.isEmpty() && m_imageQueueEmptyHysteresis)
+                m_imageQueueEmptyHysteresis->impulse();
         });
     }
 }
 
+void ImageAnalysisQueue::setImageQueueEmptyHysteresis(std::unique_ptr<PAL::HysteresisActivity>& timer)
+{
+    if (!m_imageQueueEmptyHysteresis)
```

Comment:
Something looks fishy here. There is a single ImageAnalysisQueue per Page and the hysteresis timer activity has a lambda that captures the string being searched. So, in this scenario:
1. findString("foo")
2. findString("bar")

Step 1 will create the timer and search for "foo" in a one second timer. Then step 2 will come and won't update the timer. Therefore, after I've searched for "bar", then page will asynchronously override my search and search for "foo" again. This seems wrong.

Seems we should reset the hysteresis timer but It doesn't look like we do. Did I miss anything?

Reply:
The timer only needs to be initialized and run the first time you do find-in-page on the webpage. If you load a webpage and do findString("foo"), at first only matches from the regular text on the page will appear, because the images on the page have not been analyzed yet. So to include the matches from the images, you need to analyze the images then perform findString("foo") again when the analysis completes. When you then do findString("bar"), images on the page have already been analyzed from when you did findString("foo"). There is no need to redo the image analysis, and so we don't need to worry about initializing a timer that will redo findString("bar").

Reply:
I don't think you understood my statement. What if the client searches for "foo", then for "bar" *before* your 1 second timer (which was scheduled for the "foo" search) fires?

Reply:
Ah ok, I see now. Thank you, I will look into fixing that!

Reply:
> There is no need to redo the image analysis, and so we don't need to worry about initializing a timer that will redo findString("bar").

I am not convinced this statement is correct. Pages are usually not static. Images can be dynamically added to the page at any time, which will cause `Page::didFinishLoadingImageForElement()` to get called. It will add the image to the queue and your timer WILL restart when the queue empties (again).

Reply:
Good point. Thank you!